dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -146,16 +150,70 @@ public void executeUpgrade() throws DotDataException {
         }
 
         // SCHEMA CHANGES
-        this.createWorkflowActionStepTable     (dc);
-        this.addSchemeIdColumn                 (dc);
-        //this.addRequiresCheckoutOptionColumn   (dc);
-
+        this.createWorkflowActionStepTable           (dc);
+        final boolean schemeIdColumnCreated =
+                this.addSchemeIdColumn               (dc);
+        this.addShowOnColumn                         (dc);
+        this.removeWorkflowActionStepIdWorkflowStepFK();
         // DATA CHANGES
-        this.addWorkflowActionStepData         (dc);
-        this.updateWorkflowActionData          (dc);
-        this.updateWorkflowSchemeXStructureData(dc);
+        this.addWorkflowActionStepData               (dc);
+        this.updateWorkflowActionData                (dc);
+        this.updateWorkflowSchemeXStructureData      (dc);
+
+        if (schemeIdColumnCreated) {
+            this.addNotNullConstraintShowOnColumn    (dc);
+        }
+
+
     } // executeUpgrade.
 
+    private void removeWorkflowActionStepIdWorkflowStepFK() throws DotDataException {
+
+        final DotDatabaseMetaData metaData = new DotDatabaseMetaData();
+        final ForeignKey foreignKey        = metaData.findForeignKeys
+                ("workflow_action", "workflow_step",
+                        Arrays.asList("step_id"), Arrays.asList("id"));
+
+        if (null != foreignKey) {
+
+            try {
+                Logger.info(this, "Droping the FK: " +foreignKey);
+                metaData.dropForeignKey(foreignKey);
+            } catch (SQLException e) {
+                throw new DotDataException(e.getMessage(), e);
+            } finally {
+                this.closeCommitAndStartTransaction();
+            }
+        }
+    }
+
+    private void addNotNullConstraintShowOnColumn(final DotConnect dc) throws DotHibernateException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.exception.DotHibernateException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)